File: frontend/src/services/holidayService.js
import api from './api';

export const getAllHolidays = async () => {
  const response = await api.get('/holidays');
  return response.data;
};

export const getHolidayById = async (id) => {
  const response = await api.get(`/holidays/${id}`);
  return response.data;
};

export const createHoliday = async (holidayData) => {
  const response = await api.post('/holidays', holidayData);
  return response.data;
};

export const updateHoliday = async (id, holidayData) => {
  const response = await api.put(`/holidays/${id}`, holidayData);
  return response.data;
};

export const deleteHoliday = async (id) => {
  const response = await api.delete(`/holidays/${id}`);
  return response.data;
};

export const getHolidaysInRange = async (startDate, endDate) => {
  const response = await api.get('/holidays/range', {
    params: { startDate, endDate }
  });
  return response.data;
};

export const getUpcomingHolidays = async (days = 30) => {
  const response = await api.get('/holidays/upcoming', {
    params: { days }
  });
  return response.data;
};

export const checkIsHoliday = async (date) => {
  const response = await api.get('/holidays/check', {
    params: { date }
  });
  return response.data;
};

File: backend/src/main/java/com/employeemanagement/service/HolidayService.java
package com.employeemanagement.service;

import com.employeemanagement.model.Holiday;
import java.time.LocalDate;
import java.util.List;
import java.util.Optional;

public interface HolidayService {
    Holiday createHoliday(Holiday holiday);
    Holiday updateHoliday(Long id, Holiday holiday);
    void deleteHoliday(Long id);
    Optional<Holiday> getHolidayById(Long id);
    List<Holiday> getAllHolidays();
    List<Holiday> getHolidaysInRange(LocalDate startDate, LocalDate endDate);
    boolean isHoliday(LocalDate date);
    List<Holiday> getUpcomingHolidays(int days);
}
